src/test/java/org/jabref/logic/texparser/TexParserTest.java

```diff
@@ -0,0 +1,294 @@
+    private ImportFormatPreferences importFormatPreferences;
+
+    @BeforeEach
+    void setUp() {
+        importFormatPreferences = mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS);
+    }
+
+    @AfterEach
+    void tearDown() {
+        importFormatPreferences = null;
+    }
+
+    private void testCite(String key, String citeString, boolean match) {
+        TexParserResult texParserResult = new DefaultTexParser().parse(citeString);
+        TexParserResult expectedResult = new TexParserResult();
+
+        if (match) {
+            expectedResult.getCitations().put(key, new ArrayList<>());
+            expectedResult.getCitations().get(key).add(
+                    new Citation(Paths.get("foo/bar"), 1, 0, citeString.length(), citeString));
+        }
+
+        assertEquals(expectedResult, texParserResult);
+    }
+
+    private void testMatchCite(String citeString) {
+        testCite(UNRESOLVED, citeString, true);
+    }
+
+    private void testNonMatchCite(String citeString) {
+        testCite(UNRESOLVED, citeString, false);
+    }
+
+    @Test
+    void testCiteCommands() {
+        testMatchCite("\\cite[pre][post]{UnresolvedKey}");
+        testMatchCite("\\cite*{UnresolvedKey}");
+        testMatchCite("\\parencite[post]{UnresolvedKey}");
+        testMatchCite("\\cite[pre][post]{UnresolvedKey}");
+        testMatchCite("\\citep{UnresolvedKey}");
+
+        testNonMatchCite("\\citet21312{123U123n123resolvedKey}");
+        testNonMatchCite("\\1cite[pr234e][post]{UnresolvedKey}");
+        testNonMatchCite("\\citep55{5}UnresolvedKey}");
+        testNonMatchCite("\\cit2et{UnresolvedKey}");
+    }
+
+    private void addCite(CrossingKeysResult expectedResult, String key, Path texFile, int line, int colStart, int colEnd, String lineText, boolean insert) {
+        Citation citation = new Citation(texFile, line, colStart, colEnd, lineText);
+
+        if (!expectedResult.getParserResult().getCitations().containsKey(key)) {
+            expectedResult.getParserResult().getCitations().put(key, new ArrayList<>());
+        }
+
+        if (!expectedResult.getParserResult().getCitations().get(key).contains(citation)) {
+            expectedResult.getParserResult().getCitations().get(key).add(citation);
+        }
+
+        if (insert) {
+            BibEntry clonedEntry = (BibEntry) expectedResult.getMasterDatabase().getEntryByKey(key).get().clone();
+            expectedResult.getNewDatabase().insertEntry(clonedEntry);
+        } else {
+            expectedResult.getUnresolvedKeys().add(key);
+        }
+    }
+
+    @Test
+    void testSingleFile() throws URISyntaxException, IOException {
+        InputStream originalStream = TexParserTest.class.getResourceAsStream("origin.bib");
+        Path texFile = Paths.get(TexParserTest.class.getResource("paper.tex").toURI());
+
+        try (InputStreamReader originalReader = new InputStreamReader(originalStream, StandardCharsets.UTF_8)) {
```

Comment:
Please try not to build the database based on a bib file but by hand using `BibDatabase database = ...`, `BibEntry entry = new Entry(...).withField(...)` and `database.insertEntry(...)`. Preferable you only need to this once in the `setup` method of the test class and then reuse it across the tests.